### PR TITLE
Fix up STJ comments

### PR DESCRIPTION
**Pull request by gewarren**
- Remove internal only remark
- Add see also link to new conceptual doc

### CODE REVIEW

**Comment by gewarren** [comment]
Close/reopen for build.

File: xml/System.Text.Json.Serialization.Metadata/JsonCollectionInfoValues`1.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
   <Docs>
     <typeparam name="TCollection">The collection type.</typeparam>
     <summary>Provides serialization metadata about a collection type.</summary>
-    <remarks>This API is intended for use by the output of the System.Text.Json source generator and should not be called directly.</remarks>
```
Comment (eiriktsarpalis) [inline xml/System.Text.Json.Serialization.Metadata/JsonCollectionInfoValues`1.xml:29]: This remark still holds -- this type is marked EditorBrowsable.Never and is only intended for consumption by the source generator.

File: xml/System.Text.Json.Serialization.Metadata/JsonObjectInfoValues`1.xml [hunk 2]
```diff
@@ -26,15 +26,7 @@
   <Docs>
     <typeparam name="T">The object type to serialize or deserialize.</typeparam>
     <summary>Provides serialization metadata about an object type with constructors, properties, and fields.</summary>
-    <remarks>
-      <format type="text/markdown"><![CDATA[
-
-## Remarks
```
Comment (eiriktsarpalis) [inline xml/System.Text.Json.Serialization.Metadata/JsonObjectInfoValues`1.xml:32]: Likewise, this remark is still valid.

File: xml/System.Text.Json.Serialization.Metadata/JsonParameterInfoValues.xml [hunk 3]
```diff
@@ -22,15 +22,7 @@
   </Attributes>
   <Docs>
     <summary>Provides information about a constructor parameter required for JSON deserialization.</summary>
-    <remarks>
-      <format type="text/markdown"><![CDATA[
-
-## Remarks
```
Comment (eiriktsarpalis) [inline xml/System.Text.Json.Serialization.Metadata/JsonParameterInfoValues.xml:28]: Same here.

File: xml/System.Text.Json.Serialization.Metadata/JsonTypeInfo.xml [hunk 4]
```diff
@@ -20,15 +20,7 @@
   <Interfaces />
   <Docs>
     <summary>Provides JSON serialization-related metadata about a type.</summary>
-    <remarks>
-      <format type="text/markdown"><![CDATA[
-
-## Remarks
-
-This API is intended for use by the output of the System.Text.Json source generator and should not be called directly.
```
Comment (eiriktsarpalis) [inline xml/System.Text.Json.Serialization.Metadata/JsonTypeInfo.xml:28]: Arguably, this statement is inaccurate even in .NET 6 -- `JsonTypeInfo` plays a prominent role in source generated APIs that users call into.

Comment (gewarren) [reply xml/System.Text.Json.Serialization.Metadata/JsonTypeInfo.xml:28]: These doc comments aren't versioned, so the remark will be removed for .NET 6 as well.

**Review by eiriktsarpalis — APPROVED** [review]
Thank you!

**Pull request merged**